ui/page/components/wallet_account_selector.go

```diff
@@ -334,56 +331,46 @@ func (ws *WalletAndAccountSelector) setWalletLogo(gtx C) D {
 	return inset.Layout(gtx, walletIcon.Layout24dp)
 }
 
-func (ws *WalletAndAccountSelector) ListenForTxNotifications(ctx context.Context, window app.WindowNavigator) {
-	if ws.TxAndBlockNotificationListener != nil {
-		return
+// ListenForTxNotifications listens for transaction and block updates and
+// updates the selector modal, if the modal is open at the time of the update.
+// The tx update listener MUST be unregistered using ws.StopTxNtfnListener()
+// when the page using this WalletAndAccountSelector widget is exited.
+func (ws *WalletAndAccountSelector) ListenForTxNotifications(window app.WindowNavigator) {
+	txAndBlockNotificationListener := &sharedW.TxAndBlockNotificationListener{
+		OnTransaction: func(transaction *sharedW.Transaction) {
+			// refresh wallets/Accounts list when new transaction is received
+			if ws.selectorModal != nil {
```

Comment:
I think we can do `if ws.selectorModal == nil { return }` and reduce the nested ifs.